### ISSUE TITLE
Wrong time after migration kimai v1 data

### ISSUE THREAD

**Issue by blochwitz** [kevinpapst/kimai2#507]
Hi Kevin,

after migrating data from kimai v1 the time in database table timeSheet has change (in our case all entries start 1 hour earlier).

**Comment by kevinpapst** [comment]
No. Not possible and not related to the import.
The importer only selects data from v1 - it NEVER writes!

**Comment by kevinpapst** [comment]
Are you looking at the database or the kimai frontends?

**Comment by blochwitz** [comment]
Both. I compare time entries in v2 and v1 in the database and in the frontend.

And it doesn't depend which time zone the server use - its always one hour between v1 and v2 data...

**Comment by kevinpapst** [comment]
Interesting, I haven't seen that before.
So the timestamps really change in the database for exactly 3600? 
That must be a timezone problem.
Is your cli PHP configured to use the correct timezone: `php -i |grep date.timezone` ?

**Comment by blochwitz** [comment]
Yes, its „date.timezone => Europe/Berlin => Europe/Berlin“

attached you'll find two screenshots. one of the entries in v1-db and one of the entries in v2-db

![v1](https://user-images.githubusercontent.com/42935152/51392008-048b5400-1b34-11e9-82ac-cd5fe8546d6a.png)
![v2](https://user-images.githubusercontent.com/42935152/51392012-06edae00-1b34-11e9-8178-1d7072ba4602.png)

**Comment by blochwitz** [comment]
Hi Kevin,

on my local machine i tried to import v1 data direct from our webserver into the local installed v2-db  to check if this could be a reason for changing the time entries but the command

`bin/console kimai:import-v1 "mysql://user:password@127.0.0.1:3306/database?charset=utf8" "db_prefix" "password" "country"`

does not work when a `!` is a part of the password...

**Comment by kevinpapst** [comment]
You have to url encode the password if I remember correctly: `%21`

**Comment by kevinpapst** [comment]
And I believe that 1 hour vs 2 hour:
```
$ php -r 'echo date(DATE_RFC822, 1547734500);'
Thu, 17 Jan 19 15:15:00 +0100
```
pdf
```
$ php -r 'echo date(DATE_RFC822, 1529589600);'
Thu, 21 Jun 18 16:00:00 +0200
```
is CET vs CEST: as "wintertime" is the realtime, we have CET+1 in January and CEST=CET+2 in June.

**Comment by blochwitz** [comment]
Okay, url encoding works but import direct from web server into local kima2-db wasn't a solution for time difference...

**Comment by kevinpapst** [comment]
Can you post some screenshots from Kimai v1 and Kimai v2 for these records mentioned above?

**Comment by blochwitz** [comment]
![bildschirmfoto 2019-01-21 um 16 37 06](https://user-images.githubusercontent.com/42935152/51484159-17ec2880-1d9b-11e9-8b00-b3c9b5d57a0c.png)
![bildschirmfoto 2019-01-21 um 16 38 23](https://user-images.githubusercontent.com/42935152/51484162-1884bf00-1d9b-11e9-8c59-db8ba7c79e27.png)

**Comment by kevinpapst** [comment]
I'll setup a Kimai v1 to check it in parallel. If that doesn't work I might need your database for testing.
But I'll let you know.

**Comment by kevinpapst** [comment]
You could also try to switch to the branch #372 and see if that helps to fix the problems.

**Comment by blochwitz** [comment]
I made another import of the v1-db and in v2-db the time is still wrong..

**Comment by kevinpapst** [comment]
Ok and what did you change in the meantime? More infos would help me finding the problem ;-)

**Comment by blochwitz** [comment]
Sorry, my fault. I thought you changed something while you made some tests...

**Comment by kevinpapst** [comment]
Ok, I could finally reproduce the problem, but have no idea yet where that comes from.
I'll update this issue once I know more.

**Comment by kevinpapst** [comment]
Ich habe das Problem identifiziert und muss sagen, das Du den größten Logik Fehler gefunden hast den Kimai 2 bisher hatte. Gratulation und Danke!
Wenn Du mich bei der Behebung unterstützen willst, dann wechsle auf den Branch in #372 und probiere damit den Import aus und teste im Anschluß bitte auch Kimai selbst. Aber nur für Testinstallationen, nicht für die Version die Du aktiv nutzt! Die Umstellung ist noch nicht final. 
In den Benutzereinstellungen kann man jetzt die Timezone einstellen, auch das sollte mal getestet werden. Ich mache das natürlich auch selber, aber mir ist das Problem bisher auch nicht im vollen Umfang klar gewesen.

**Comment by blochwitz** [comment]
Hi Kevin,

i'm really sorry but in the „prepare-0.8“-branch your fixes of timezone doesn't work proper. I made some tests and there is still a difference of one hour: in the list and calendar view all entries I migrate from v1 are wrong (-1 hour). in the edit view of an entry the correct time will display.

**Comment by kevinpapst** [comment]
Did you import from v1 after switching to the branch of before?

**Comment by blochwitz** [comment]
I made both

**Comment by kevinpapst** [comment]
Ok, I could imagine that importing after switching to the branch might fail (I didn't try that, will do later).

But if you imported the times before, there should be no problems, I tested that a dozen times. 
Did you execute the convert command? 
Please share more infos what you did and in which order, otherwise I can't help.

The edit and create timesheet form is displaying a wrong time, I am currently working on a fix.

**Comment by blochwitz** [comment]
I didn’t execute the convert command cause I import from v1 into a fresh database (like you describe ist in #372)

In my case the time of an imported entry is displaying in the edit view like it must but the same entry is wrong in the list and calendar view.

**Comment by kevinpapst** [comment]
Ok, that means might assumption is correct: the importer is now broken.
As the list is displaying the correct time and the edit form is displaying a converted time. As I found out in #554 
LOL, fucking timezones ... I'll fix that and create a new PR.

**Comment by blochwitz** [comment]
Perfect and thanks!

**Comment by kevinpapst** [comment]
Can you please try #555 - I hope that finally fixes everything...
And thanks for all the support, your time and feedback!

**Comment by blochwitz** [comment]
Yes, the time of entries is fine now.
